feat(en): Verify priority txs

Deniallugo opened the pull request:
## What ❔

Verify L1txs during the EN sync

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

Deniallugo committed:
feat(en): verify txs

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Add retries

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
use only v29 version

Signed-off-by: Danil <deniallugo@gmail.com>

perekopskiy commented on core/node/node_sync/src/external_io.rs:
```diff
@@ -375,11 +389,40 @@ impl StateKeeperIO for ExternalIO {
+                    for i in 0..=max_attempts {
+                        if let Err(err) = self
+                            .txs_verifier
+                            .verify_transaction(l1_tx, protocol_version)
```
why? I meant do not starting any block executing before validation passes

Deniallugo replied:
you mean check all txs in a batch all at once, right?

Deniallugo commented on core/node/node_sync/src/external_io.rs:
```diff
@@ -491,6 +534,153 @@ impl StateKeeperIO for ExternalIO {
+                None,
+                None,
+            )
+            .from_block((tx.common_data.eth_block - 1).into())
+            .to_block((tx.common_data.eth_block + 1).into())
```
could we just keep sending eth_block?

perekopskiy replied:
maybe, I don't remember why it was deprecated..

zkzoomer committed:
bump contracts

Deniallugo committed:
Support eth block for consensus

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
fix: fix force deployments (#4424)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into deniallugo-check-priority-txs

zkzoomer committed:
bump contracts

zkzoomer committed:
regenesis
